assets/js/googlesitekit/datastore/site/first-party-mode.js

```diff
@@ -0,0 +1,224 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { isPlainObject } from 'lodash';
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import {
+	commonActions,
+	createRegistrySelector,
+	combineStores,
+	createReducer,
+} from 'googlesitekit-data';
+import { CORE_SITE } from './constants';
+import { createFetchStore } from '../../data/create-fetch-store';
+
+const SET_FIRST_PARTY_MODE_ENABLED = 'SET_FIRST_PARTY_MODE_ENABLED';
+
+const settingsReducerCallback = createReducer(
+	( state, firstPartyModeSettings ) => {
+		state.firstPartyModeSettings = firstPartyModeSettings;
+	}
+);
+
+const fetchGetFirstPartyModeSettingsStore = createFetchStore( {
+	baseName: 'getFirstPartyModeSettings',
+	controlCallback: () =>
+		API.get( 'core', 'site', 'fpm-settings', undefined, {
+			useCache: false,
+		} ),
+	reducerCallback: settingsReducerCallback,
+} );
+
+const fetchSaveFirstPartyModeSettingsStore = createFetchStore( {
+	baseName: 'saveFirstPartyModeSettings',
+	controlCallback: ( { settings } ) => {
+		return API.set( 'core', 'site', 'fpm-settings', { settings } );
+	},
+	reducerCallback: settingsReducerCallback,
+	argsToParams: ( settings ) => {
+		const { isEnabled } = settings || {};
+		return { settings: { isEnabled } };
+	},
+	validateParams: ( { settings } ) => {
+		invariant(
+			isPlainObject( settings ),
+			'settings must be a plain object.'
+		);
+
+		invariant(
+			typeof settings.isEnabled === 'boolean',
+			'isEnabled must be a boolean.'
+		);
+
+		invariant(
+			Object.keys( settings ).length === 1,
+			'settings must have only the `isEnabled` property.'
+		);
+	},
+} );
+
+const baseInitialState = {
+	firstPartyModeSettings: undefined,
+};
+
+const baseActions = {
+	/**
+	 * Saves the first-party mode settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	*saveFirstPartyModeSettings() {
+		const { select } = yield commonActions.getRegistry();
+		const settings = select( CORE_SITE ).getFirstPartyModeSettings();
+
+		return yield fetchSaveFirstPartyModeSettingsStore.actions.fetchSaveFirstPartyModeSettings(
+			settings
+		);
+	},
+
+	/**
+	 * Sets the first-party mode enabled status.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object} Redux-style action.
+	 */
+	setFirstPartyModeEnabled() {
+		return {
+			type: SET_FIRST_PARTY_MODE_ENABLED,
```

Comment:
Can we return a `payload` property?
```suggestion
			type: SET_FIRST_PARTY_MODE_ENABLED,
			payload: { isEnabled },
```

Reply:
Thanks for spotting that, @hussain-t! I've updated it to provide a payload :+1: